### PR TITLE
Acceleration Get IP

### PR DESCRIPTION
**Pull request by helen-frank**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Acceleration Get IP
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kwok/controllers/node_controller.go [hunk 1]
```diff
@@ -484,7 +484,12 @@ func (c *NodeController) computePatch(node *corev1.Node, tpl string) ([]byte, er
 	})
 }
 
-// putNodeInfo puts node info (HostIPs)
+// putNodeInfo puts node info (HostIPs and PodCIDRs)
+// HostIps: either a single "primary IP" for the node in a single-stack cluster, or a dual-stack
+// pair of IPs in a dual-stack cluster (for nodes that actually have dual-stack IPs).
+// Among other things, the IPs returned from this function are used as the `.status.PodIPs`
+// values for host-network pods on the node, and the first IP is used as the `.status.HostIP`
+// for all pods on the node.
```
[the remark this answers is not in the document]
Comment (helen-frank) [reply pkg/kwok/controllers/node_controller.go:492]: My original consideration was that the offline environment was not convenient to view.

[the remark this answers is not in the document]
Comment (helen-frank) [reply pkg/kwok/controllers/node_controller.go:492]:
```suggestion
// HostIps: either a single "primary IP" for the node in a single-stack cluster, or a dual-stack
// pair of IPs in a dual-stack cluster (for nodes that actually have dual-stack IPs).
// Among other things, the IPs returned from this function are used as the `.status.PodIPs`
// values for host-network pods on the node, and the first IP is used as the `.status.HostIP`
// for all pods on the node. Reference to https://github.com/kubernetes/kubernetes/blob/1d02d014e8c1f0de84b0b58b2165548182815320/pkg/util/node/node.go#L67-L104
```

[the remark this answers is not in the document]
Comment (wzshiming) [reply pkg/kwok/controllers/node_controller.go:492]: That's ok

[the remark this answers is not in the document]
Comment (helen-frank) [reply pkg/kwok/controllers/node_controller.go]: ok

File: pkg/kwok/controllers/node_controller.go [hunk 2]
```diff
@@ -484,9 +484,34 @@ func (c *NodeController) computePatch(node *corev1.Node, tpl string) ([]byte, er
...
 
-// putNodeInfo puts node info (HostIPs)
+// putNodeInfo puts node info (HostIPs and PodCIDRs)
 func (c *NodeController) putNodeInfo(node *corev1.Node) {
-	nodeIPs := []string{}
+	nodeIPs := GetNodeHostIPs(node)
+	hostIps := make([]string, 0, len(nodeIPs))
+	for i := range nodeIPs {
+		hostIps = append(hostIps, nodeIPs[i].String())
+	}
```
[the remark this answers is not in the document]
Comment (helen-frank) [reply pkg/kwok/controllers/node_controller.go]: ok

File: pkg/kwok/controllers/node_controller.go [hunk 3]
```diff
@@ -484,9 +485,33 @@ func (c *NodeController) computePatch(node *corev1.Node, tpl string) ([]byte, er
...
+	c.nodesSets.Store(node.Name, nodeInfo)
+}
+
+// GetNodeHostIPs returns the provided node's IP(s); either a single "primary IP" for the
+// node in a single-stack cluster, or a dual-stack pair of IPs in a dual-stack cluster
+// (for nodes that actually have dual-stack IPs). Among other things, the IPs returned
+// from this function are used as the `.status.PodIPs` values for host-network pods on the
+// node, and the first IP is used as the `.status.HostIP` for all pods on the node.
+// Refer to https://github.com/kubernetes/kubernetes/blob/1d02d014e8c1f0de84b0b58b2165548182815320/pkg/util/node/node.go#L67-L104
```
Comment (wzshiming) [inline pkg/kwok/controllers/node_controller.go:512]:
```suggestion
// Copy from https://github.com/kubernetes/kubernetes/blob/1d02d014e8c1f0de84b0b58b2165548182815320/pkg/util/node/node.go#L67-L104
```

**Review by wzshiming — COMMENTED** [review]
Others LGTM

File: pkg/kwok/controllers/node_controller.go [hunk 4]
```diff
@@ -484,9 +485,33 @@ func (c *NodeController) computePatch(node *corev1.Node, tpl string) ([]byte, er
...
+func GetNodeHostIPs(node *corev1.Node) []net.IP {
```
Comment (wzshiming) [inline pkg/kwok/controllers/node_controller.go:513]:
```suggestion
func getNodeHostIPs(node *corev1.Node) []net.IP {
```

Comment (helen-frank) [reply pkg/kwok/controllers/node_controller.go]: 👌

**Pull request merged**